### PR TITLE
Equilibrium Index of an Array

### LINKED ISSUE #8395: To Find Equilibrium Index of an Array
### Description

C_CPP

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by yellowberard**
<hr>

## Description 📜
Explanation about how to find equilibrium index of an array.
Fixes #8395 

<hr>

## Type of change 📝

- [x] Doc#umentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x]  C/#CPP
<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

<hr>

### CODE REVIEW

**Comment by yellowberard** [comment]
@vasu-1 Please review my PR.

File: C_CPP/Data Structures/Array/Equlibrium_index.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# Equilibrium Index of a Array
+
+Here we will see how to find a equlibrium index in a array.
+
+Equlibrium index means that index i such that the sum all the lower indexes is same to the sum of the higher indexes.
+
+I hve given a example to under stand it better.
```
Comment (Suvraneel) [inline C_CPP/Data Structures/Array/Equlibrium_index.md:7]:
```suggestion
I have given a example to under stand it better.
```

File: C_CPP/Data Structures/Array/Equlibrium_index.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# Equilibrium Index of a Array
+
+Here we will see how to find a equlibrium index in a array.
+
+Equlibrium index means that index i such that the sum all the lower indexes is same to the sum of the higher indexes.
```
Comment (Suvraneel) [inline C_CPP/Data Structures/Array/Equlibrium_index.md:5]:
```suggestion
Equilibrium index means that index i such that the sum all the lower indexes is same to the sum of the higher indexes.
```

**Review by Suvraneel — CHANGES_REQUESTED** [review]
Do a spell-check before a mentor reviews this... Mistake in file name too

**Comment by yellowberard** [comment]
@Suvraneel I have made the changes. Sorry for spelling mistakes.

File: C_CPP/Data Structures/Array/Equilibrium_index.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+Equilibrium index means that index i such that the sum of all the lower indexes is same to the sum of the higher indexes.
+
+I have given a example to understand it better.
+
+## Example
+
+**Input**: Arr[] = {-7, 1, 5, 2, -4, 3, 0}
+
+**Output**: 3
```
Comment (Suvraneel) [inline C_CPP/Data Structures/Array/Equilibrium_index.md:13]:
```suggestion
# Equilibrium Index of a Array
```suggestion
# Equilibrium Index of a Array

Here we will see how to find a equilibrium index in a array.

Equilibrium index means that index i such that the sum of all the lower indexes is same to the sum of the higher indexes.

I have given a example to understand it better.

## Example

**Input**: Arr[] = {-7, 1, 5, 2, -4, 3, 0}  
**Output**: 3
```
Here we will see how to find a equilibrium index in a array.  
Equilibrium index means that index i such that the sum of all the lower indexes is same to the sum of the higher indexes.  
I have given a example to understand it better.

## Example  
**Input**: Arr[] = {-7, 1, 5, 2, -4, 3, 0}  
**Output**: 3  
```

File: C_CPP/Data Structures/Array/Equilibrium_index.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+```
+
+### Time Complexity
+
+O(n)
+
+### Space Complexity
+
+O(1)
```
Comment (Suvraneel) [inline C_CPP/Data Structures/Array/Equilibrium_index.md:74]:
```suggestion
### Time Complexity  
O(n)  
### Space Complexity  
O(1)
```

**Review by Suvraneel — CHANGES_REQUESTED** [review]
@yellowberard  No need to be Sorry... 😅 
These are not wrong...
But the objective is to make you understand opensource formatting standards...

Here, you've used 2 newline characters for each new paragraph...
But line cunts in commit matters...
So, instead of pressing Enter key twice, you need to add 2 spaces at End of line & press Enter

File: C_CPP/Data Structures/Array/readme.md [hunk 5]
```diff
@@ -9,3 +9,4 @@
 - [Matrix Multiplication](./Matrix_multiplication/Matrix_multiplication.md)
 - [Subarray & Subsequence](./Subarray&Subsequence.md)
 - [ARRAY BAR CHART](./ARRAY%20BAR%20CHART/bargraph.md)
+- [Equilibrium Index of a Array](Equlibrium_index.md)
```
Comment (Suvraneel) [inline C_CPP/Data Structures/Array/readme.md:12]:
```suggestion
- [Equilibrium Index of a Array](Equilibrium_index.md)
```
Fix Link

**Review by Suvraneel — CHANGES_REQUESTED** [review]

**Comment by yellowberard** [comment]
@Suvraneel I have made the changes.

**Comment by Suvraneel** [comment]
@yellowberard  Noice work 🚀  LGTM ❤️

**Comment by yellowberard** [comment]
> @yellowberard Noice work 🚀 LGTM ❤️

Thank you

**Pull request merged**